workflows/genomes_generation.nf

```diff
@@ -101,10 +101,10 @@ workflow GGP {
     // ---- combine data for reads and contigs pre-processing ---- //
     groupReads = { meta, assembly, fq1, fq2 ->
         if (fq2 == []) {
-            return tuple(meta, assembly, [fq1])
+            return tuple(meta + [single_end: true], assembly, [fq1])
         }
         else {
-            return tuple(meta, assembly, [fq1, fq2])
+            return tuple(meta + [single_end: false], assembly, [fq1, fq2])
         }
     }
     assembly_and_runs = Channel.fromSamplesheet("samplesheet", header: true, sep: ',').map(groupReads) // [ meta, assembly_file, [raw_reads] ]
```

Comment:
I am wondering if in this function fq1 indeed corresponds to read_1.fq.gz and fq2 corresponds to read_2.fq.gz? You expect to have that order on the csv input file right? If this is the case, I think you don't need to sort the files evaluating the name of the file in L14 of modules/local/align_bwa/main.nf, which is a tricky piece of code